core.cpp

```diff
@@ -66,6 +66,10 @@ struct ByteArray Mat_ToBytes(Mat m) {
     return toByteArray(reinterpret_cast<const char*>(m->data), m->total() * m->elemSize());
 }
 
+struct DataPtr Mat_DataPtr(Mat m) {
```

Comment:
Seems like we could reuse ByteArray struct here instead of creating a new type. Basically just do what you already implemented but with ByteArray, and since Mat_DataPtr does not perform the memcpy, it does the right thing.